libraries/apollo-normalized-cache-api-incubating/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/internal/OptimisticCache.kt

```diff
@@ -14,115 +14,130 @@ import kotlin.reflect.KClass
 
 @ApolloInternal
 class OptimisticCache : NormalizedCache() {
+  private val lock = Lock()
   private val recordJournals = mutableMapOf<String, RecordJournal>()
 
   override fun loadRecord(key: String, cacheHeaders: CacheHeaders): Record? {
-    return try {
-      val nonOptimisticRecord = nextCache?.loadRecord(key, cacheHeaders)
-      nonOptimisticRecord.mergeJournalRecord(key)
-    } catch (ignore: Exception) {
-      null
+    return lock.read {
+      try {
+        val nonOptimisticRecord = nextCache?.loadRecord(key, cacheHeaders)
+        nonOptimisticRecord.mergeJournalRecord(key)
```

Comment:
Should this be a `lock.write {}`? (I expect `mergeJournalRecord` to modify data)

Reply:
Mmm nevermind, the returned `nonOptimisticRecord` isn't shared so all good 👍